pontoon/base/models.py

```diff
@@ -2872,28 +2874,139 @@ def serialize(self):
 
 
 class TranslationMemoryEntryManager(models.Manager):
-    def minimum_levenshtein_ratio(self, text, min_quality=0.7):
+    def get_levenshtein_sql(self, expr):
+        """
+        Parametrize shared sql query for levenshtein ratio.
+
+        :param str expr: SQL expression (e.g. model field)
+        :return: a string with SQL query which calculates the levenshtein ratio.
+        """
+        levenshtein_ratio_equation = """(
+                (char_length({expr}) + %s - levenshtein({expr}, %s, 1, 2, 2))::float /
+                (char_length({expr}) + %s)
+            )"""
+        return levenshtein_ratio_equation.format(expr=expr)
+
+    def postgres_levenshtein_ratio(self, locale, text, min_quality, min_dist, max_dist):
+        """
+        Filter TranslationMemory entries fully in PostgreSQL.
+        `levenshtein` function is provided by `fuzzystrmatch` module.
+        All strings are initially pre-filtered with min_dist and max_dist.
+
+        :arg Locale locale:
+        :arg str text: reference string to search in  TM
+        :arg float min_quality: minimal quality of a levenshtein ratio
+        :arg int min_dist: minimum length distance from a text string
+        :arg int max_dist: maximum length distance from a text string
+        :return: TranslationMemory Entries enriched with the quality metric.
+        """
+        if len(text) > 255 or min_dist > 255 or max_dist > 255:
+            raise ValueError("Can't search for strings longer than 255 characters.")
+
+        entries = (
+            self.extra(
+                where=[
+                    '(CHAR_LENGTH(source) BETWEEN %s AND %s)',
+                    self.get_levenshtein_sql('source') + ' > %s'
+                ],
+                params=(min_dist, max_dist, len(text), text, len(text), min_quality),
```

Comment:
I'm not sure if I can pass named parameters here, because Django doesn't support them or I can't find it :)
https://github.com/django/django/blob/master/django/db/models/sql/query.py#L1877
I have an idea to create queryset filters to remove this `extra` calls, I'll try to do this tomorrow.